Katello/smart_proxy_container_gateway
# Fixes #31566 - support all search methods

This PR enables v1 ping and search for backwards compatibility.  Docker, but not Podman, seems to want to use the v1 APIs.

To test:

1) Set up a smart proxy with Pulp 3 and this plugin installed
2) Push a couple of (no-auth-needed) container images to your smart proxy
3) Make sure ```docker search `hostname`/``` works
4) Try ```docker search `hostname`/<substring>``` to test filtering out images
5) To test image count limiting, try ```curl "https://`hostname`/v1/search?q=&n=<limit value>"```
